core/src/main/scala/com/fortysevendeg/kafka/config/implicits/ConsumerConfiguration.scala

```diff
@@ -0,0 +1,92 @@
+      .join(ConfigValueDecoder[Int]("max.poll.interval.ms").optional)
+      .join(ConfigValueDecoder[Int]("max.poll.records").optional)
+      .join(ConfigValueDecoder[java.util.List[String]]("partition.assignment.strategy").optional)
+      .join(ConfigValueDecoder[Int]("receive.buffer.bytes").optional)
+      .join(ConfigValueDecoder[Int]("request.timeout.ms").optional)
+      .join(ConfigValueDecoder[String]("sasl.jaas.config").optional)
+      .join(ConfigValueDecoder[String]("sasl.kerberos.service.name").optional)
+      .join(ConfigValueDecoder[String]("sasl.mechanism").optional)
+      .join(ConfigValueDecoder[String]("security.protocol").optional)
+      .join(ConfigValueDecoder[Int]("send.buffer.bytes").optional)
+      .join(ConfigValueDecoder[java.util.List[String]]("ssl.enabled.protocols").optional)
+      .join(ConfigValueDecoder[String]("ssl.keystore.type").optional)
+      .join(ConfigValueDecoder[String]("ssl.protocol").optional)
+      .join(ConfigValueDecoder[String]("ssl.provider").optional)
+      .join(ConfigValueDecoder[String]("ssl.truststore.type").optional)
+
+  implicit val lowPriorityConsumerConfig: Decoder[Config, Map[String, Any]] =
+    ConfigValueDecoder[Int]("auto.commit.interval.ms").optional
+      .join(ConfigValueDecoder[Boolean]("check.crcs").optional)
+      .join(ConfigValueDecoder[String]("client.id").optional)
+      .join(ConfigValueDecoder[Int]("fetch.max.wait.ms").optional)
+      .join(ConfigValueDecoder[java.util.List[String]]("interceptor.classes").optional)
+      .join(ConfigValueDecoder[Long]("metadata.max.age.ms").optional)
+      .join(ConfigValueDecoder[java.util.List[String]]("metric.reporters").optional)
+      .join(ConfigValueDecoder[Int]("metrics.num.samples").optional)
+      .join(ConfigValueDecoder[String]("metrics.recording.level").optional)
+      .join(ConfigValueDecoder[Long]("metrics.sample.window.ms").optional)
+      .join(ConfigValueDecoder[Long]("reconnect.backoff.max.ms").optional)
+      .join(ConfigValueDecoder[Long]("reconnect.backoff.ms").optional)
+      .join(ConfigValueDecoder[Long]("retry.backoff.ms").optional)
+      .join(ConfigValueDecoder[String]("sasl.kerberos.kinit.cmd").optional)
+      .join(ConfigValueDecoder[Long]("sasl.kerberos.min.time.before.relogin").optional)
+      .join(ConfigValueDecoder[Double]("sasl.kerberos.ticket.renew.jitter").optional)
+      .join(ConfigValueDecoder[Double]("sasl.kerberos.ticket.renew.window.factor").optional)
+      .join(ConfigValueDecoder[java.util.List[String]]("ssl.cipher.suites").optional)
+      .join(ConfigValueDecoder[String]("ssl.endpoint.identification.algorithm").optional)
+      .join(ConfigValueDecoder[String]("ssl.keymanager.algorithm").optional)
+      .join(ConfigValueDecoder[String]("ssl.secure.random.implementation").optional)
+      .join(ConfigValueDecoder[String]("ssl.trustmanager.algorithm").optional)
+
+  implicit val consumerConfig: Decoder[Config, Map[String, AnyRef]] = {
+    for {
```

Comment:
this can be better expressed in Applicative style with the cartesian builder since none of this operations are sequential, ordered or interdependent from the results of the previous one in order to create the final map.